src/nspawn/nspawn.c

```diff
@@ -2594,9 +2594,24 @@ static int inner_child(
                 return -ESRCH;
         }
 
-        r = mount_systemd_cgroup_writable("", arg_unified_cgroup_hierarchy);
-        if (r < 0)
-                return r;
+	if (cg_ns_supported()) {
+		r = unshare(CLONE_NEWCGROUP);
+		if (r < 0)
+			return log_error_errno(errno, "Failed to unshare cgroup namespace");
+		r = mount_cgroups(
+				"",
+				arg_unified_cgroup_hierarchy,
+				arg_userns_mode != USER_NAMESPACE_NO,
+				arg_uid_shift,
+				arg_uid_range,
+				arg_selinux_apifs_context);
+		if (r < 0)
+			return r;
```

Comment:
```
Jun 28 13:12:02 adt systemd[1]: Starting Container c1...
Jun 28 13:12:02 adt systemd-nspawn[1485]: Selected user namespace base 84410368 and range 65536.
Jun 28 13:12:02 adt systemd-nspawn[1485]: mount(/var/lib/machines/c1/sys/fs/selinux) failed, ignoring: No such file or directory
Jun 28 13:12:02 adt systemd-nspawn[1485]: mount(/var/lib/machines/c1/sys/fs/selinux) failed, ignoring: Invalid argument
Jun 28 13:12:02 adt systemd-nspawn[1485]: Timezone Etc/UTC does not exist in container, not updating container timezone.
Jun 28 13:12:02 adt systemd-nspawn[1485]: Failed to determine if /sys/fs/cgroup is already mounted: No such file or directory
Jun 28 13:12:02 adt systemd-nspawn[1485]: Child died too early.
Jun 28 13:12:02 adt systemd[1]: systemd-nspawn@c1.service: Main process exited, code=exited, status=1/FAILURE
Jun 28 13:12:02 adt systemd[1]: Failed to start Container c1.
Jun 28 13:12:02 adt systemd[1]: systemd-nspawn@c1.service: Unit entered failed state.
Jun 28 13:12:02 adt systemd[1]: systemd-nspawn@c1.service: Failed with result 'exit-code'.
```
I think
```c
r = path_is_mount_point(cgroup_root, AT_SYMLINK_FOLLOW);
if (r < 0)
	return log_error_errno(r, "Failed to determine if /sys/fs/cgroup is already mounted: %m");
```
doesn't work in the inner child (after `mount_move_root`)
Seems like we should check `/sys/fs/cgroup` in the `outer_child` and pass the result of the check to the `inner_child`.